src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -731,7 +731,7 @@ protected function mergeWheresToHas(Builder $hasQuery, Relation $relation)
      * Get the "has relation" base query instance.
      *
      * @param  string  $relation
-     * @return \Illuminate\Database\Eloquent\Builder
+     * @return \Illuminate\Database\Eloquent\Relations\Relation
```

Comment:
`Builder::has()` [expects](https://github.com/laravel/framework/blob/5.1/src/Illuminate/Database/Eloquent/Builder.php#L589) this to be a relation instance when passet to [`Builder::addHasWhere`](https://github.com/laravel/framework/blob/5.1/src/Illuminate/Database/Eloquent/Builder.php#L696)